diff --git a/src/arguments.py b/src/arguments.py
index 729b842..dc50a44 100644
--- a/src/arguments.py
+++ b/src/arguments.py
@@ -1,5 +1,5 @@
 """Common arguments for train and evaluation for RPMNet"""
-import argparse
+import argparse  # The argparse module makes it easy to write user-friendly command-line interfaces
 
 
 def rpmnet_arguments():
diff --git a/src/train.py b/src/train.py
index 05ab0b3..6d9c78a 100644
--- a/src/train.py
+++ b/src/train.py
@@ -4,19 +4,19 @@ Example usage:
     python train.py --noise_type crop
     python train.py --noise_type jitter --train_batch_size 4
 """
-from collections import defaultdict
-import os
+from collections import defaultdict  # Dict subclass that calls a factory function to supply missing values
+import os  # Provides functions for interacting with operating system
 import random
-from typing import Dict, List
+from typing import Dict, List # Generic version of dict and list, useful for annotating return types
 
 from matplotlib.pyplot import cm as colormap
 import numpy as np
-import open3d  # Ensure this is imported before pytorch
-from tensorboardX import SummaryWriter
-import torch
-import torch.nn as nn
-import torch.utils.data
-from tqdm import tqdm
+import open3d  # Ensure this is imported before pytorch   # Deals with 3D data
+from tensorboardX import SummaryWriter # Provide visualisation of traning progress of neural network
+import torch                         # use for deep learning aplications using GPU and CPU
+import torch.nn as nn                # use for creating and training of neural network
+import torch.utils.data             # Represents a Python iterable over a dataset,
+from tqdm import tqdm          #Use for creating Progress Meters or Progress Bars
 
 from arguments import rpmnet_train_arguments
 from common.colors import BLUE, ORANGE
